pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BucketDelayedDeliveryTracker.java

```diff
@@ -91,9 +98,66 @@ public BucketDelayedDeliveryTracker(PersistentDispatcherMultipleConsumers dispat
         this.sharedBucketPriorityQueue = new TripleLongPriorityQueue();
         this.immutableBuckets = TreeRangeMap.create();
         this.snapshotSegmentLastIndexTable = HashBasedTable.create();
-        this.numberDelayedMessages = 0L;
         ManagedCursor cursor = dispatcher.getCursor();
         this.lastMutableBucket = new MutableBucket(cursor, bucketSnapshotStorage);
+        this.numberDelayedMessages = recoverBucketSnapshot();
+    }
+
+    private long recoverBucketSnapshot() throws RuntimeException {
```

Comment:
Is it better to add a method `CompletableFuture<Void> initialize()` to the `DelayedDeliveryTracker` to avoid blocking calls when creating the `BucketDelayedDeliveryTracker` instance

Reply:
It seems that he needs to block `initialize` method, when using `DelayedDeliveryTrackerFactory` to create `DelayedEdeliverytracker`

Reply:
Ok, we can just keep it here for now.